src/nupic/types/Sdr.hpp

```diff
@@ -239,6 +239,11 @@ class SparseDistributedRepresentation : public Serializable
      */
     const UInt &size = size_;
 
+    /**
+     * Change the dimensions of the SDR.  The total size must not change.
+     */
+    void reshape(const std::vector<UInt> &dimensions) const;
```

Comment:
consider also returning the SDR? (*this), to allow chaining?

Reply:
I'd rather do this in a separate PR, and change all of the C++ methods at once.  I will open an issue for this.

Reply:
See issue #487